### ISSUE TITLE
Mac Version is 32-bit

### ISSUE THREAD

**Issue by Avakining** [jaquadro/NBTExplorer#83]
The Mac version, 2.0.3, is 32-bit, and as such will not be supported on the next version of macOS, macOS Catalina. 
Additionally, the Mac version is not available to download from github, only from minecraftforum -> hocuspocus.taloncrossing.com/rii/NBTExplorer-Mac-2.0.3.zip

**Comment by FireBall1725** [comment]
+1 for Mac 64bit support

**Comment by ghost** [comment]
+1

**Comment by HunterAhlquist** [comment]
+1,000,000

**Comment by CyanideCarrot** [comment]
@jaquadro any update on this? The new macOS is out now

**Comment by jaquadro** [comment]
I haven't had a working toolchain to build for mac for a number of years.

The landscape has changed since I essentially lost access monomac in 2013 or 14.  Microsoft acquired Xamarin and the tools and SDKs appear to be free now.  But I don't really have the time (or possibly hardware) to try and get this build working again.  If a mac developer is interested in trying to restart this, please reach out to me.

**Comment by FireBall1725** [comment]
@jaquadro I am a developer who uses macOS, I am willing to give it a shot to build it for the newer mac if you can give me a few pointers, not used to Xamarin and Visual Studio.

**Comment by FireBall1725** [comment]
Also, if it is something that building can be automated, CircleCI offers a free build agent for open source projects running macOS (https://circleci.com/open-source/)

**Comment by jaquadro** [comment]
I'd start with trying to build anything with VS and Xamarin for mac, like a sample program.  If you can get that far, then it might be as simple as loading the mac proj file for NBTExplorer, and replacing the monomac assembly reference with whatever the new xamarin mac one is.

That's probably wishful thinking, especially if there were changes in the xamarin API.  But who knows.

**Comment by NBSgamesAT** [comment]
A 64-bit version for Catalina would really be cool. I do have to say tho that help should have been searched as the "This Application is not optimised for your Mac" message appeared the first time. But when no one's opening an issue, how should you know?. I am a Mac developer but since Im just getting serious with mac specific stuff so I don't think I'm much help.

With that said: I'm wishing FireBall1725 good luc

**Comment by sportshead** [comment]
@FireBall1725 @jaquadro Any update on the 64-bit version?

**Comment by shai** [comment]
+1

**Comment by shai** [comment]
Also, I found this today, might be a good alternative until this issue is resolved: [webNBT](https://irath96.github.io/webNBT/)

**Comment by jaquadro** [comment]
Has anyone tried using the windows/linux version (via the posted zip file) on Mono?  I double-checked the build configuration and it's not targeted at a specific CPU architecture.

It's not as pretty as the old mac version, but that version also doesn't support one of the more recently added tag types.

**Comment by SullyTheUnusual** [comment]
> Has anyone tried using the windows/linux version (via the posted zip file) on Mono? I double-checked the build configuration and it's not targeted at a specific CPU architecture.
> 
> It's not as pretty as the old mac version, but that version also doesn't support one of the more recently added tag types.

Just tried and failed.  Looks like the Carbon driver for Mono hasn't been ported to 64-bit yet.  Would love an update from @FireBall1725, if they're still working on it.

**Comment by FireBall1725** [comment]
@SullyTheUnusual: Back in October I tried to figure out Visual Studio and Xamarin but didn't make much progress, had a lot of issues importing the project since the Xamarin used in the project isn't available anymore. I ended up having to put all that on hold because I ended up moving back in December and have been pretty busy.

**Comment by SullyTheUnusual** [comment]
@FireBall1725 Thanks for your efforts, and the quick reply!  :)

**Comment by MrPickles2009** [comment]
@FireBall1725 do you have a fork that you've made progress on with the mac version? I can help you out if you want

**Comment by FireBall1725** [comment]
Hey @MrPickles2009, I believe i just cloned it, and never forked it. Even if I did fork it, I literally made no progress, I tried a few things I read online to try to convert it to the newer Xamarin stuff, but just never made any progress that actually got the project to open. Everything kept showing errors and pointing me back to missing libraries, no matter what I edited couldn't point it to the new ones.

**Comment by MrPickles2009** [comment]
@FireBall1725  i'll take a stab at this after i finish work

**Comment by brothanb** [comment]
I just installed mono on my Mac and the 2.8.0 binary fired up under Wine 4.0.1 (which I already had installed) without any issues.

**Comment by MrPickles2009** [comment]
@brothanb and you’re running macos 10.15?

**Comment by brothanb** [comment]
I’m still running 10.14.

**Comment by Avakining** [comment]
@brothanb well in 10.15 all 32-bit apps don't work at all.

**Comment by Avakining** [comment]
As far as I can tell, this is due to the windows NBTExplorer using WinForms, which mono uses Carbon to replicate. Carbon was deprecated several years ago and removed in 10.15, so everything crashes and burns if you try to use it.

There's [ongoing debate](https://github.com/mono/mono/issues/6701) on how to fix it, but nobodies actually managed to make a workable version yet.

The best way to get NBTExplorer to work on mac is to either completely start over (probably in XCode) and make a whole new app or update the Windows version to not use WinForms.

**Comment by Elhison** [comment]
A bit off-topic: where is the download link for macOS? I can't find it anywhere.

**Comment by 5ynt4x-3rr0r** [comment]
> A bit off-topic: where is the download link for macOS? I can't find it anywhere.

@CottonTheButton On the minecraftfourms page, but it's outdated


> Has anyone tried using the windows/linux version (via the posted zip file) on Mono? I double-checked the build configuration and it's not targeted at a specific CPU architecture.
> 
> It's not as pretty as the old mac version, but that version also doesn't support one of the more recently added tag types.

@jaquadro I tried it, as others said Carbon is deprecated so it didn't work. Since the source code is available the community can compile it, maybe someone with a bit more expertise (and free time) than me can compile it and publish it.

**Comment by brothanb** [comment]
> Since the source code is available the community can compile it, maybe someone with a bit more expertise (and free time) than me can compile it and publish it.

I've tried compiling it, but there is a dependency on a 32-bit only module that doesn't have a 64-bit equivalent.   I don't remember what it was and haven't revisited this in a while due to the lack of free time myself.

**Comment by DumbGameMaker** [comment]
> 
> 
> I just installed mono on my Mac and the 2.8.0 binary fired up under Wine 4.0.1 (which I already had installed) without any issues.

on 10.15 you can try compiling wine with the `--use-win64` (i think) option in `./configure`, I've done it before, but when I tried to run NBTExplorer with it I got a nice error message. wonder why, it's probably not because 10.15 removed support for 32-bit programs. y'know this is why I like windows: backwards- and forwards-compatibility.

**Comment by MrPickles2009** [comment]
> y'know this is why I like windows: backwards- and forwards-compatibility.

@DumbGameMaker this is true but people aren't going to stop using mac any time soon so devs have to adapt...

**Comment by Shock9616** [comment]
Any update on this yet? I have needed NBTExplorer multiple times recently and haven't been able to

**Comment by RealRTTV** [comment]
use another editor?
